common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -60,6 +60,24 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
 
   }
 
+  def cleanAmpSoundcloud(document: Document) = {
+    document.getElementsByClass("element-audio").filter { audioElement: Element =>
+      audioElement.getElementsByTag("iframe").length != 0
+    }.foreach { audioElementWithIframe: Element =>
+      audioElementWithIframe.getElementsByTag("iframe").foreach { iframeElement: Element =>
+        val soundcloudUrl = iframeElement.attr("src")
+        val pattern = "%2Ftracks%2F(\\d+)".r
+        val trackId = pattern.findAllIn(soundcloudUrl).matchData.map { matches => matches.group(1) }.mkString
+        val soundcloud = document.createElement("amp-soundcloud")
+        soundcloud.attr("data-trackid", trackId)
+        soundcloud.attr("data-visual", "true")
+        soundcloud.attr("height", "300") // height is necessary if data-visual == true
+        audioElementWithIframe.appendChild(soundcloud)
+        iframeElement.remove()
```

Comment:
there's a lot of heavy lifting in this process that's a little distracting, so think about having a `getIdFromElement` and a `createAmpElement` function and calling one then the other and then do the append/remove as a final step